src/Logger/RulesLog.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Logger\RulesLog.
+ */
+
+namespace Drupal\rules\Logger;
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\Core\Logger\LogMessageParserInterface;
+use Drupal\Core\Logger\RfcLoggerTrait;
+use Drupal\rules\Event\SystemLoggerEvent;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Dispatches new logger-items as SystemLoggerEvent.
+ */
+class RulesLog implements LoggerInterface {
+  use RfcLoggerTrait;
+
+  /**
+   * An EventDispatcherInterface instance.
+   */
+  protected $dispatcher;
+
+  /**
+   * The message's placeholders parser.
+   *
+   * @var \Drupal\Core\Logger\LogMessageParserInterface
+   */
+  protected $parser;
+
+  /**
+   * Constructs a RulesLog object.
+   *
+   * @param EventDispatcherInterface $dispatcher
+   *   An EventDispatcherInterface instance.
+   * @param \Drupal\Core\Logger\LogMessageParserInterface $parser
+   *   The parser to use when extracting message variables.
+   */
+  public function __construct(EventDispatcherInterface $dispatcher, LogMessageParserInterface $parser) {
+    $this->dispatcher = $dispatcher;
+    $this->parser = $parser;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @todo: discuss if the subject-format. Currently it is a parsed and flatten array.
+   */
+  public function log($level, $message, array $context = array()) {
+    // Remove any backtraces since they may contain an unserializable variable.
+    unset($context['backtrace']);
+
+    // Convert PSR3-style messages to SafeMarkup::format() style, so they can be
+    // translated too in runtime.
+    $message_placeholders = $this->parser->parseMessagePlaceholders($message, $context);
+
+    $subject = array(
+      'uid' => $context['uid'],
```

Comment:
Should also be fixed with TypedData LoggerEntry
https://www.drupal.org/node/2625238